src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -60,7 +59,7 @@ public static bool IsRightSideOfColonColon(this ExpressionSyntax expression)
             => expression.IsParentKind(SyntaxKind.AliasQualifiedName, out AliasQualifiedNameSyntax aliasName) && aliasName.Name == expression;
 
         public static bool IsRightSideOfDot(this ExpressionSyntax name)
```

Comment:
this guy is reasonable.  he's not asking "is this the name of a member access expression", merely "am i on the right side of a dot in my parent node".  the name does not imply any particular node shape.